### PR TITLE
feat(docs): add global search functionality in docs

### PR DESCRIPTION
**Pull request by anuradha9712**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: docs/src/components/GlobalSearch/index.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  )
+}
+
+const Search = ({parentRef}) => {
+  const [query, setQuery] = useState(null);
+  const [searchResult, setSearchResult] = useState([]);
+  const [openPopover, setOpenPopover] = useState(false);
+  const searchList = useSearchItems();
+
+  // useEffect(() => {
```
Comment (xlreon) [inline docs/src/components/GlobalSearch/index.js]: remove this if not needed or else add appropriate comments if we intend to use it in future

File: docs/src/components/GlobalSearch/index.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    element.addEventListener('keydown', handleInstantSearch);
+
+    return () => {
+      element.removeEventListener('keydown', handleInstantSearch);
+    }
+  }, []);
+
+  const handleInstantSearch = (e) => {
+    const inputField = searchRef.current;
+    if (e.code === "KeyS" && inputField !== document.activeElement) {
```
Comment (xlreon) [inline docs/src/components/GlobalSearch/index.js]: We can use e.key here for better readability

File: docs/src/components/GlobalSearch/index.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            icon="search"
+            name="input"
+            autoComplete="off"
+            placeholder="Search components, patterns.."
+            onChange={({ target }) => handleSearchQuery(target)}
+          />
+        }
+      >
+        {
+          !query &&
```
Comment (xlreon) [inline docs/src/components/GlobalSearch/index.js]: We can separate out the conditional markups into a function

File: docs/src/components/GlobalSearch/index.js [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        break;
+
+      default:
+        break;
+    }
+  })
+
+  return (
+    <div>
+      {components.length > 0 && <ShowResults query={query} name="Components" list={components} />}
```
Comment (xlreon) [inline docs/src/components/GlobalSearch/index.js:93]: We can combine all these into one

File: docs/src/components/GlobalSearch/index.js [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  searchList.forEach((item) => {
+    const slugPath = item.node.slug.split('/');
+    let category = '';
+    if (slugPath[0] === 'mobile') {
+      category = slugPath[1];
+    } else {
+      category = slugPath[0];
+    }
+
+    switch (category) {
```
Comment (xlreon) [inline docs/src/components/GlobalSearch/index.js:61]: convert this switch case to object map

File: docs/src/components/GlobalSearch/search.css [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  max-height: 400px;
+  width: 450px;
+}
+
+.search-result-entry:hover{
+  background-color: var(--secondary-lightest);
+}
+
+.search-result-link{
+  text-decoration: none !important;
```
Comment (xlreon) [inline docs/src/components/GlobalSearch/search.css]: Why are we using !important here?

**Review by xlreon — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**